gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -361,7 +393,95 @@ abstract class KspTaskJvm : KotlinCompile(KotlinJvmOptionsImpl()), KspTask {
                 kotlinCompile.compilerArgumentsContributor
             }
         )
-        this.useClasspathSnapshot.value(false).disallowChanges()
+
+        isIntermoduleIncremental = project.findProperty("ksp.incremental.intermodule")?.toString().toBoolean()
+        if (isIncremental && isIntermoduleIncremental) {
+            val classStructureIfIncremental = project.configurations.create("_classStructure${name}")
```

Comment:
You can change this to be a detached configuration i.e use `project.configurations.detachedConfiguration`.

If you want to create a named one (as you are doing right now), you need to create it at the time the task is registered, otherwise it may happen too late because of lazy task configuration.

Reply:
Did you mean this? 8b9dc928be0fe610adae3a5cddad2f168ff19285

Reply:
Exactly 👍